bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Composite.java

```diff
@@ -1581,14 +1581,25 @@ LRESULT WM_PAINT (long wParam, long lParam) {
 				GC paintGC = null;
 				Image image = null;
 				if ((style & (SWT.DOUBLE_BUFFERED | SWT.TRANSPARENT)) != 0) {
-					image = new Image (display, width, height);
 					paintGC = gc;
-					gc = new GC (image, paintGC.getStyle() & SWT.RIGHT_TO_LEFT);
-					GCData gcData = gc.getGCData ();
-					gcData.uiState = data.uiState;
-					gc.setForeground (getForeground ());
-					gc.setBackground (getBackground ());
-					gc.setFont (getFont ());
+					int originalStyle = gc.getStyle();
+					ImageGcDrawer drawer = new ImageGcDrawer() {
+					    @Override
+					    public void drawOn(GC gc, int iWidth, int iHeight) {
+					    		GCData gcData = gc.getGCData ();
+							gcData.uiState = data.uiState;
+							gc.setForeground (getForeground ());
+							gc.setBackground (getBackground ());
+							gc.setFont (getFont ());
+					    }
+
+					    @Override
+					    public int getGcStyle() {
+					        return  originalStyle & SWT.RIGHT_TO_LEFT;
+					    }
+					};
+					image = new Image (display, drawer, width, height);
+					gc = new GC(image, originalStyle & SWT.RIGHT_TO_LEFT);
```

Comment:
Are you sure that this works as expected? You now initialize the GC inside the `ImageGcDrawer` and then create a _new_ one that is used throughout the rest of the method, which does not adhere to the settings inside `ImageGcDrawer`. At least, it would be my expectation that those two GCs behave completely independent. In my understanding, the code you move inside the `ImageGcDrawer` now equals to a no-op, doesn't it?